### PR TITLE
Bugfix: make sure that every destroyed plugin handle is 'detached'

### PR DESCRIPTION
**Pull request by SamyCookie**
Sometimes, when sessions are destroyed, the `detached` property is not correctly set in plugin handle.
This patch attempts to fix this bad behaviour

### CODE REVIEW

**Comment by lminiero** [comment]
That's not the proper fix. It assumes the client is always the one detaching the handle, which is not true: plugins can be detached programmatically by plugins, meaning users will receive a "detached" event without sending a detach themselves.

**Comment by lminiero** [comment]
Edit: checking the code, though, `plugin.detach()` in the event will invoke `destroyHandle`, so we should be all set in that case too. Sorry forthe noise!

**Comment by SamyCookie** [comment]
> Edit: checking the code, though, `plugin.detach()` in the event will invoke `destroyHandle`, so we should be all set in that case too. Sorry forthe noise!

Indeed, my patch extends the `detached` behaviour in the detach/destroy client side, but keep the previous behaviour in server side as well (via the event).

**Comment by SamyCookie** [comment]
Also, I think that the `.ondetached()` callback associated to the `detached` property should be moved in `destroyHandle` at the same place, but it was not my bug...

**Comment by lminiero** [comment]
Merging, thanks :+1:

**Comment by SamyCookie** [comment]
the same ;) Thank you for merging this !
